### PR TITLE
serial/deserial for playerlist, serial for card from 20_ui because i am terrible at git

### PR DESCRIPTION
**Pull request by michael-s-myer**
ui files to be added to integ

### CODE REVIEW

File: dialog.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+class Dialog(object):
+    """
+    Pop-up that contains a message and has two buttons.  Initiated with references
+    to the actions that should be taken for each button.
+    """
+
+    message = None  # a string representing the message to display
+    left_button = None  # a button representing a command
+    right_button = None # a button representing a command
+
```
Comment (patmay) [inline dialog.py]: The above lines create class/static attrs (which are separate from the instance attrs you reference below with self._attr. Please remove these, or put them in a comment.

File: dialog.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    def __init__(self, message, left_button, right_button):
+        """Constructor for Dialog which accepts two Button objects and a String"""
+        self.__message = message
```
Comment (patmay) [inline dialog.py]: Please change to single underscores for instance attr names.

File: game_app.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+class GameApp(object):
+    """
+    Triggered by controllers to update after a move or suggestion has been asked.
+    Can also show new views due to user action (asking for a suggestion).  Extends UserInterface.
+    """
+
+    note_card_view = None   # two-dimensional array of button objects representing the note card information
+    note_card = None    # a reference to the current note_card
+    note_card_controller = None    # a reference to the NoteCardController to send clicks
```
Comment (patmay) [inline game_app.py]: Please remove these

File: picker.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    choices = None  # an array of toggle objects that contains the choices
+    key = None
+
+    def __init__(self, choices):
+        """Constructor that accepts an array of card objects to initialize the Picker object"""
+        self.__choices = choices
+
+    def disable_button(self, key):
+        """Accepts an integer identifying the button to disable"""
+        return True
```
Comment (patmay) [inline picker.py]: Not sure these need to return anything, but no harm done

**Comment by patmay** [comment]
Please remove the class attrs in each file, and change the instance attrs to single underscores.

**Comment by yo09975** [comment]
Didn't see the removal of class scoped variables as recommended by @patmay -- moved back to In Progress.

**Comment by michael-s-myer** [comment]
is there a conflict still?  any help is welcome

File: src/card.py [hunk 5]
```diff
@@ -42,3 +42,9 @@ def get_type(self) -> CardType:
         """Returns an Enum that represents the type of the Card"""
         # print('get_type method in Card class')
         return self._card_type
+
+    def serialize(self, payload):
+        """serialize card_id to a JSON file"""
+        card = {}
+        card['card_id'] = self.get_id()
+        return json.dumps(payload)
```
Comment (yo09975) [inline src/card.py:50]: You serialize a Card, not payload.  You deserialize the payload, back into a Card. Also, you're missing the other member variables of the Card.

Take a look at how the serialize for the other classes works.

Comment (patmay) [reply src/card.py:50]: This is taken care of in issue #33 / PR #67

**Comment by yo09975** [comment]
What's the status of this? Do you need help refactoring? It looks like the work for 20_ui is being worked in tandem by others (drop the PR for the ui changes??) and the serial/deserial should get its own PR. Let me know how I can help.

**Comment by yo09975** [comment]
Rejected.

**Pull request closed without merging**